### PR TITLE
Save color palettes

### PR DESCRIPTION
**Pull request by IvanUkhov**
We would like to take a look at the color data to see if we can do anything useful with it.

### CODE REVIEW

File: internal/font_metadata.py [hunk 1]
```diff
@@ -156,18 +156,10 @@ def _read_color(ttf):
...
-        numPalettes = 0
-        numPaletteEntries = 0
-
         if "CPAL" in ttf:
-            numPaletteEntries = ttf["CPAL"].numPaletteEntries
-            numPalettes = len(ttf["CPAL"].palettes)
-
-        return {
-            "formats": t,
-            "numPalettes": numPalettes,
```
Comment (IvanUkhov) [inline internal/font_metadata.py:168]: This and next can be derived from `palettes`. So redundant.

File: internal/font_metadata_test.py [hunk 2]
```diff
@@ -14,6 +14,7 @@ def test_read_metadata_color():
 
     assert actual == expected
 
+
```
Comment (IvanUkhov) [inline internal/font_metadata_test.py:17]: Due to Black.

File: internal/font_metadata.py [hunk 3]
```diff
@@ -156,18 +156,10 @@ def _read_color(ttf):
...
-            "numPaletteEntries": numPaletteEntries,
-        }
+            return {"formats": t, "paletts": ttf["CPAL"].palettes}
```
Comment (bramstein) [inline internal/font_metadata.py:160]: typo?

**Comment by pmeenan** [comment]
Is there a risk of the data size exploding (i.e. how big can the palettes get)? Should the summary data be maintained and only the first N palette entries included?

File: internal/font_metadata.py (hunk 3 above)
Comment (IvanUkhov) [reply internal/font_metadata.py]: Uh, fixed. Thank you!

**Comment by IvanUkhov** [comment]
That is a valid concern. For non-emoji fonts, it is probably very few. But for those with emojis, it can be quite a bit. I was testing once for Noto Color Emoji, and it was 5895 entries:

https://github.com/bodoni/font/blob/main/tests/palettes.rs#L25

**Comment by IvanUkhov** [comment]
I have gathered some statistics on the number of palettes and color entries, and in my sample, the median of the former was four and of the latter two. So I set the limit to 10. Should be sufficient.

**Comment by pmeenan** [comment]
LGTM. Ready for it to be merged?

**Comment by pmeenan** [comment]
Going ahead with the merge since the crawl will be kicking off in ~30 minutes

**Pull request merged**